### PR TITLE
#1659 visual indication when tabbing through delete and add

### LINKED ISSUE #1659: Finance Fixes - You Should Be Able to Tab Through Product Section
### Description

Right now you can tab through all the buttons on the product table in a reimbursement request create/edit view, but the delete button and add buttons don't give any indication when they're selected. Ideally they should have the same slightly different look they have when you hover over them.

### Acceptance Criteria

- [ ] When you tab to a delete button, it has the same look as when you hover over it
- [ ] When you tab to an add product button, it has the same look as when you hover over it

### Proposed Solution
Look at the MUI docs to see how to change the look of something while focused on (which I believe are state these buttons are in when you tab to them)

### PR DESCRIPTION
**Pull request by msereeyothin**
## Changes

There is now a visual indication when tabbing through delete and add button of the products of the reimbursement request. 

## Screenshots

add product tabbed
delete button tabbed

Closes #1659

### CODE REVIEW

**Review by callmejim1226 — CHANGES_REQUESTED** [review]
Great start!

When I ran the local Finishline app in dark mode, I checked the 'Add Product' button to get a white background after clicking it. 

Can we fix this and make sure the button looks good in the dark mode too?

image

**Comment by msereeyothin** [comment]
It should now work in light and dark mode. However, the background color while hovering of the 'add product' button is slightly different to match the color of the focus.

File: src/frontend/src/pages/FinancePage/ReimbursementRequestForm/ReimbursementProductTable.tsx [hunk 1]
```diff
@@ -162,7 +166,10 @@ const ReimbursementProductTable: React.FC<ReimbursementProductTableProps> = ({
                     sx={{
                       margin: '4px',
                       '&:focus': {
-                        backgroundColor: '#fdf8f8'
+                        backgroundColor: hoverColor
```
Comment (callmejim1226) [inline src/frontend/src/pages/FinancePage/ReimbursementRequestForm/ReimbursementProductTable.tsx:169]: I think we can remove `&focus` prop from `sx` prop to remove the different color background on 'add product' button. It got removed in my local app after I removed `&focus` prop.

Comment (msereeyothin) [reply src/frontend/src/pages/FinancePage/ReimbursementRequestForm/ReimbursementProductTable.tsx:169]: Removing `&:focus` from the sx prop makes it so there is no indication when we tab through to the add product button. I think that would go against the acceptance criteria for this ticket?

Comment (callmejim1226) [reply src/frontend/src/pages/FinancePage/ReimbursementRequestForm/ReimbursementProductTable.tsx:169]: I see. According to the ticket description, your point sounds right. However, I don't get the point of us having a highlight show up on the button after we click on `ADD PRODUCT` and `delete icon`.

@RChandler234 Let us know if this is what you expected for this ticket.

Comment (RChandler234) [reply src/frontend/src/pages/FinancePage/ReimbursementRequestForm/ReimbursementProductTable.tsx:169]: you can make the onclick method for add product and it'll fix both problems:
```
(e) => {
                      appendProduct({
                        wbsNum: validateWBS(key),
                        name: '',
                        cost: 0
                      });
                      e.currentTarget.blur();
                    }
```

**Review by RChandler234 — CHANGES_REQUESTED** [review]

**Review by callmejim1226 — APPROVED** [review]
LGTM! Great job!

**Review by RChandler234 — APPROVED** [review]
LGTM 🔥

**Pull request merged**